### PR TITLE
Unbreak xvfb

### PR DESCRIPTION
**Pull request by agx**
x11: make sure event source is drained
    
 Otherwise running under Xvfb will not deliver any events. This results in e.g. weston-info reporting a 0x0 window size (which results in all sorts of problems) plus some other minor cleanups.

### CODE REVIEW

**Comment by emersion** [comment]
`wl_event_loop_fd_func_t` does say the return value should be:

> If the event source is registered for re-check with wl_event_source_check(): 0 for all done, 1 for needing a re-check. If not registered, the return value is ignored and should be zero.

However `x11_event` always returns 0, which means we always say "this doesn't need to re-check" (same as not calling `wl_event_source_check`). So I don't quite understand what's happening.

(BTW, we have a headless backend)

**Comment by agx** [comment]
Hi,
On Wed, Jul 11, 2018 at 08:51:39AM -0700, emersion wrote:
>    wl_event_loop_fd_func_t does say the return value should be:
> 
>      If the event source is registered for re-check with
>      wl_event_source_check(): 0 for all done, 1 for needing a re-check. If
>      not registered, the return value is ignored and should be zero.
> 
>    However x11_event always returns 0, which means we always say "this
>    doesn't need to re-check" (same as not calling wl_event_source_check). So
>    I don't quite understand what's happening.

See ``wl_event_loop_dispatch()``. It makes sure that we're dispatched
again after idle sources. Not sure why regular x.org does not need this
but having Xvfb working is a nice thing.

> 
>    (BTW, we have a headless backend)

Yeah, I know but I want some X11 things here to synthesize input events,
etc.

**Comment by emersion** [comment]
>See ``wl_event_loop_dispatch()``. It makes sure that we're dispatched again after idle sources. Not sure why regular x.org does not need this but having Xvfb working is a nice thing.

Oh, right. Thanks for the explanation.

>Yeah, I know but I want some X11 things here to synthesize input events, etc.

(You can also do this with the headless backend - but ofc yeah, we want the X11 backend to work in Xvfb)

**Comment by emersion** [comment]
Thanks!

File: backend/x11/output.c [hunk 1]
```diff
@@ -47,8 +47,16 @@ static bool output_set_custom_mode(struct wlr_output *wlr_output,
...
 	const uint32_t values[] = { width, height };
-	xcb_configure_window(x11->xcb_conn, output->win,
+	xcb_void_cookie_t cookie = xcb_configure_window_checked(x11->xcb_conn, output->win,
 		XCB_CONFIG_WINDOW_WIDTH | XCB_CONFIG_WINDOW_HEIGHT, values);
+
+	xcb_generic_error_t *error;
+	if ((error = xcb_request_check(x11->xcb_conn, cookie))) {
+		wlr_log(WLR_ERROR, "Could not set window size to %dx%d\n", width, height);
+		free(error);
+		return false;
```
Comment (psychon) [inline backend/x11/output.c]: Should this also print information about what the error was?
Also, what kind of errors do you expect? According to https://www.x.org/releases/X11R7.5/doc/x11proto/proto.html (search for `ConfigureWindow`), the possible errors are Match, Value, and Window:

- BadMatch cannot happen here (happen when setting the border width of an InputOnly window or changing the stacking order in an incorrect way).
- BadValue can easily be checked against before sending the broken request (happens when setting width or height to zero).
- BadWindow... well, you have worse problems if you got your own window ID wrong.

So basically, this change introduces an unnecessary round-trip to the X11 server, I think.
